src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -91,11 +92,23 @@ public void CloneTo(X509Certificate2Collection collection)
                 return;
             }
 
+            var loadedCerts = new HashSet<X509Certificate2>();
+
             foreach (string filePath in Directory.EnumerateFiles(_storePath, PfxWildcard))
             {
                 try
                 {
-                    collection.Add(new X509Certificate2(filePath));
+                    var cert = new X509Certificate2(filePath);
+
+                    // If we haven't already loaded a cert .Equal to this one, copy it to the collection.
```

Comment:
Looks like Issuer + Serial.  Though I suppose we could make the HashSet use RawData instead if you're wondering if you should express concern